### PR TITLE
Adding PG16 and reworking to minimal Alpine image

### PR DESCRIPTION
**Pull request by BigAlRender**
We received a ticket from a customer who wanted PG16 backup support. Unfortunately, just adding "16" to the script wasn't cutting it.

Purely as a suggestion, I've created a reworked version based on Alpine. 

It makes the image much smaller (~85%):
```
REPOSITORY              TAG               IMAGE ID       CREATED          SIZE
pg_dump_alpine          latest            8467c6648381   17 minutes ago   178MB
pg_dump_original        latest            ef5ea1b1433d   22 minutes ago   1.15GB
```

However, it's not all upsides:
- It does require a little more config, `ALPINE_VERSION` along with `POSTGRES_VERSION` (values in README)
- Alpine doesn't seem to have an `apk` package for PG11.

### CODE REVIEW

**Comment by Weixuanf** [comment]
@BigAlRender is this planned to be merged?

**Comment by Weixuanf** [comment]
if not i guess i'll just fork this as my repo and set 

POSTGRES_VERSION = 16

ALPINE_VERSION = 3.20

and it should be good to go?

**Comment by gertjanjansen** [comment]
Thanks @BigAlRender for preparing this. I was able to set up backups for pg 16 with the fork. Maybe it's good to update the documentation here until this is merged: https://render.com/docs/backup-postgresql-to-s3. There will be more and more users with pg >15, right?

File: README.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ This repo can be used to deploy a [Cron Job](https://render.com/docs/cronjobs) o
...
 
+You'll need to set corresponding OS and Postgres versions:
+
+| `POSTGRES_VERSION` | `ALPINE_VERSION` |
+| :----------------: | :--------------: |
+| 16                 | 3.20             |
+| 15                 | 3.20             |
+| 14                 | 3.20             |
+| 13                 | 3.19             |
+| 12                 | 3.18             |
```
Comment (breel-render) [inline README.md:15]: As 12 is EOL, we should skip this row like 11

```suggestion
```

**Pull request merged**